lib/stellar_core_commander/docker_process.rb

```diff
@@ -72,7 +72,8 @@ def cleanup
     Contract None => Any
     def dump_database
       Dir.chdir(working_dir) do
-        `docker exec #{state_container_name} pg_dump -U #{database_user} --clean --no-owner #{database_name}`
+        host_args = "-H tcp://#{docker_host}:#{docker_port}" if host
+        `docker #{host_args} exec #{state_container_name} pg_dump -U #{database_user} --clean --no-owner #{database_name}`
```

Comment:
I'm kind of tempted to change this to `IO.popen` to share logic with the `docker` function, but that would likely require revamping how we handle process spawning and output gathering entirely.

So going to defer that until a later PR and allow a little duplication here.